src/msg/async/frames_v2.h

```diff
@@ -63,6 +63,29 @@ struct segment_t {
   __le16 alignment;
 } __attribute__((packed));
 
+struct onwire_segment_t {
+  // crypto-processed segment can be expanded on-wire because of:
+  //  * padding to achieve CRYPTO_BLOCK_SIZE alignment,
+  //  * authentication tag. It's appended at the end of message.
+  //    See RxHandler::get_extra_size_at_final().
+  __le32 onwire_length;
+
+  struct ceph::msgr::v2::segment_t logical;
+} __attribute__((packed));
```

Comment:
what's the purpose of separating onwire_segment_t from segment_t?

Reply:
Not needed anymore as auth tag has been conceptually moved from *last segment* into *epilogue* -- `next_payload_len` for `Message` handling is abandoned. I think this allows round up to crypto block size (`::get_onwire_size`) just when allocating the buffer for storing physical (read: encrypted) representation of segment's data. Will clean this up.

Reply:
Dropped.